var/spack/repos/builtin/packages/py-sqlalchemy/package.py

```diff
@@ -22,3 +23,12 @@ class PySqlalchemy(PythonPackage):
     depends_on('py-pytest@2.5.2:3.9.0,3.9.3:', type='test')
     depends_on('py-mock', type='test')
     depends_on('py-pytest-xdist', type='test')
+
+    variant('mysql', default='none', description='MySQL support',
+            values=('none', 'mysqlclient', 'pymysql'), multi=False)
+    depends_on('py-mysqlclient', when='mysql=mysqlclient', type=('build', 'run'))
+    depends_on('py-pymysql',     when='mysql=pymysql',     type=('build', 'run'))
+
+    variant('postgresql', default='none', description='PostgreSQL support',
+            values=('none', 'psycopg2'))
+    depends_on('py-psycopg2',    when='postgresql=psycopg2')
```

Comment:
I'm inclined to use 3+ Boolean variants, since that's how pip would handle these extras. Alternatively, we could make a single multi-valued variant that allows any of these backends. See `py-matplotlib backend=...` for an example of how this would work.

Reply:
I've decided to go with multi-valued variant